src/Features/CSharp/Portable/UsePatternMatching/CSharpIsAndCastCheckWithoutNameDiagnosticAnalyzer.cs

```diff
@@ -57,21 +59,34 @@ private void SyntaxNodeAction(SyntaxNodeAnalysisContext context)
             var syntaxTree = semanticModel.SyntaxTree;
 
             // "x is Type y" is only available in C# 7.0 and above.  Don't offer this refactoring
-            // in projects targetting a lesser version.
+            // in projects targeting a lesser version.
             if (((CSharpParseOptions)syntaxTree.Options).LanguageVersion < LanguageVersion.CSharp7)
             {
                 return;
             }
 
-            var root = syntaxTree.GetRoot(cancellationToken);
             var isExpression = (BinaryExpressionSyntax)context.Node;
 
-            var workspace = (context.Options as WorkspaceAnalyzerOptions)?.Services.Workspace;
+            var options = context.Options as WorkspaceAnalyzerOptions;
+            var workspace = options?.Services.Workspace;
             if (workspace == null)
             {
                 return;
             }
 
+            var optionSet = options.GetDocumentOptionSetAsync(syntaxTree, cancellationToken).GetAwaiter().GetResult();
```

Comment:
I think this proves definite benefit of having a special analyzer in Roslyn.Diagnostics.Analyzers (roslyn-analyzers repo) which is dedicated analyzer package for Roslyn.sln. Let me file an issue.

Reply:
agreed :)

Reply:
https://github.com/dotnet/roslyn-analyzers/issues/2508